### PR TITLE
Weekly Patch Release v1.5.1

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do? 

Contains all bugfixes to be released next week. As bugfixes get merged to master, they will get cherry-picked into this branch by the individual author (core) or on-caller.

For staff/core: The new [release guide is here](https://www.notion.so/gridai/Release-Guide-ab132bf94f304056afc8ff992304c18f).

```bash
gh pr list -s merged -S 'merged:2021-11-01T18:30:00.000Z..2021-11-10T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "1.5.x") or (.milestone.title == null)) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

```
2021-11-09T14:58:31Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10434 7fb277f260bf91137af1ac599ed2751bbf1d49e3 Resolve workers being forcelly deleted with `persistent_workers=True`
2021-11-09T14:51:53Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10432 f1882aa69fe5eaa7bc9a8027ca7e184c17d05502 [Fault Tolerance] Don't check the len of a dataset, but its instance.
2021-11-09T14:21:01Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10429 aaa6aa75e9e10de8857324e2142e137e7d7a43b2 Fix converting only float type tensors in Lite
2021-11-09T11:23:27Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10409 21eafafcb0951ac8032de779b1d70c7de0af4c0b disable step logging in epoch hooks
2021-11-08T11:28:59Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10376 613aa0951416dde0ff8bb8dfa78f7808e83e7ff2 Revert part of #10279
2021-11-08T10:36:35Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10388 89e1360e756d5a697863e5aa28379e1cf587c422 Fix pickling error with CSVLogger
2021-11-08T02:05:44Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10359 bdc24e558a075acf5e71c957f24eb05ea7b25f28 Only import PostLocalSGD related modules when it's needed
2021-11-05T17:31:45Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10334 45f6a3b1758f88af7fd776915539800cbc0137a9 Fix DataLoader inspection and re-instantiation in Lite
2021-11-04T17:26:25Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10269 939a861853059d4a632843dbbf83e9ab802ad424 Update Python testing
2021-11-04T11:46:57Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10345 ce3e63262a7247c1111375b28c90bdd83b42bf3f Fix failure when `DataLoader(batch_size=None)` is passed
2021-11-03T11:18:10Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10316 93caa7cda9277f86f17b1eab0793019aee8d40c2 Fix `apply_to_collection(defaultdict)`
2021-11-02T21:36:18Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10307 4cd7e77ad2471379eaf768c20d8a3284aeb8b0b5 Add note for PyCharm users on configuration for `RichProgressBar`
2021-11-02T17:43:55Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10311 72288b2b32b5466aed531ac7e6b885ecf7da7b0e update ref to 1.5 as stable

```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

### CODE REVIEW

**Comment by awaelchli** [comment]
Some fixes that could potentially be included today: #10429, https://github.com/PyTorchLightning/pytorch-lightning/pull/10428, https://github.com/PyTorchLightning/pytorch-lightning/pull/10374/

**Review by rohitgr7 — APPROVED** [review]
don't forget to update the description once finalized :)

**Review by kaushikb11 — APPROVED** [review]
New Chapter begins! 🥳

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]
just not sure if we really want to update all CI do newer versions of Python, I would probably rather stay conservative, and as we have not updated it before v1.5 then do not update bugfix branch either...

**Comment by awaelchli** [comment]
This would concern the commit 4f6dcb9, right? Requesting comment from @carmocca.

**Comment by carmocca** [comment]
It's fine to keep, no versions were added or removed, the jobs were just reorganized.
